### PR TITLE
Add PegasusIO

### PR DESCRIPTION
**Pull request by yihming**
This is a package manipulating multi-modal omics data, with auxiliary functionality on count matrix aggregation and quality-control. And it's also the IO module of Pegasus, a single-cell data analysis package in Python. We feel like that we need to first add this package to Bioconda before adding Pegasus as a whole.

The recipe has been locally tested in "Bootstrap" method. Please let us know if any further modification is needed.

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by yihming** [comment]
@BiocondaBot please add label

File: recipes/pegasusio/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    version: {{ version }}
+
+source:
+    url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+    sha256: "95f678bbed68b12da8f05304738b633ff111d9043df2f5753636081350cb5a9b"
+
+
+build:
+    number: 0
+    entry_points:
```
Comment (dpryan79) [inline recipes/pegasusio/meta.yaml:15]: Please put the following line in the `build:` section:

    skip: True  # [py27]

File: recipes/pegasusio/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        - pegasusio = pegasusio.__main__:main
+    script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv"
+
+requirements:
+    build:
+        - {{ compiler('c') }}
+    host:
+        - python >=3.6
```
Comment (dpryan79) [inline recipes/pegasusio/meta.yaml:23]: Please remove `>=3.6`

File: recipes/pegasusio/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        - cython
+        - setuptools_scm
+        - pip
+    run:
+        - numpy
```
Comment (dpryan79) [inline recipes/pegasusio/meta.yaml:28]: It's implied that you need python, but please make that explicit by adding ` - python` to the run: section.

**Review by dpryan79 — CHANGES_REQUESTED** [review]
See comments. BTW, we can't edit your PR (perhaps your organization prevents that), otherwise I'd have just directly made the suggested changes.

**Comment by yihming** [comment]
@dpryan79 Thank you so much for your suggestions! I've updated my recipe.

BTW, may I ask the meaning of `skip: True  # [py27]`? In specific, would "# [py27]" be parsed so that the build only skip Python 2? Thanks!

**Comment by dpryan79** [comment]
Yes, `skip: True  # [py27]` means exactly that python 2.7 builds will be skipped. Since this requires compiling against specific versions of python, we can't restrict the python version in the host or run section (doing so results in a build for only python 3.8).

**Comment by dpryan79** [comment]
@bioconda-bot please fetch artifacts

**Review by dpryan79 — APPROVED** [review]

**Comment by dpryan79** [comment]
@bioconda-bot please merge

**Pull request merged**